timeseries/src/autogluon/timeseries/models/registry.py

```diff
@@ -0,0 +1,62 @@
+from abc import ABCMeta
+from dataclasses import dataclass
+from typing import Dict
+
+
+@dataclass
+class ModelRecord:
+    model_class: type
+    default_priority: int
+
+
+class ModelRegistry(type):
```

Comment:
Nit: Should we align this with [Tabular](https://github.com/autogluon/autogluon/blob/master/tabular/src/autogluon/tabular/register/_model_register.py#L14)? I think `Registry` is the more precise term, I wonder if we should also change `ModelRegister` -> `ModelRegistry` on the tabular side.

Reply:
I'm ok with changing to `ModelRegistry` so long as we do it before the 1.3 release. Feel free to open a PR for it

Reply:
Created #5107 to align